src/orion/core/worker/experiment.py

```diff
@@ -150,6 +150,17 @@ def __setstate__(self, state):
         for key, value in state.items():
             setattr(self, key, value)
 
+    def _restore_storage(self):
+        self._storage = get_storage()
```

Comment:
I think you should add a comment here explaining why we need to do this. (the weird evc issue with pickling)